### PR TITLE
Facade_Engine: Handle nulls

### LINKED ISSUE #2477: Facade Engine - handle nulls
For the following methods. FYI @enarhi I'm doing this as I raise the issue in case you're working on this engine.

 - [x] BH.Engine.Facade.Query.Depth(FrameEdgeProperty frameEdgeProp)
 - [x] BH.Engine.Facade.Query.MaterialComposition(Panel panel)
 - [x] BH.Engine.Facade.Query.MaterialComposition(Opening opening)
 - [x] BH.Engine.Facade.Query.MaterialComposition(FrameEdge frameEdge)
 - [x] BH.Engine.Facade.Query.SimpleGeometry(FrameEdgeProperty frameEdgeProp)
 - [x] BH.Engine.Facade.Query.SolidVolume(Panel panel)
 - [x] BH.Engine.Facade.Query.SolidVolume(Opening opening)
 - [x] BH.Engine.Facade.Query.SolidVolume(FrameEdge frameEdge)
 - [x] BH.Engine.Facade.Query.Width(FrameEdgeProperty frameEdgeProp)
 - [x] BH.Engine.Facade.Query.IsAdjacent(Line curve1, Line curve2, Double tolerance)
 - [x] BH.Engine.Facade.Query.IsAdjacentApprox(Line curve1, Line curve2, Double tolerance)
 - [x] BH.Engine.Facade.Query.PrimaryPropertyName(FrameEdge frameEdge)
 - [x] BH.Engine.Facade.Query.PrimaryPropertyName(IEdge edge)
 - [x] BH.Engine.Facade.Query.PrimaryPropertyName(Panel panel)
 - [x] BH.Engine.Facade.Query.PrimaryPropertyName(Opening opening)
 - [x] BH.oM.Facade.Elements.Opening.Opening(IEnumerable<ICurve> edges, IConstruction construction, FrameEdgeProperty frameEdgeProperty, String name)
 - [x] BH.oM.Facade.Elements.Panel.Panel(PlanarSurface surface, Construction construction, FrameEdgeProperty frameEdgeProperty, PanelType panelType, IConstruction openingConstruction, and 2 more inputs)

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #2477 

### Additional comments
Fixed some minor compliance as I went through @enarhi FYI.

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@BHoMBot check null-handling

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check dataset-compliance

File: Facade_Engine/Query/PrimaryPropertyName.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public static string IPrimaryPropertyName(this IElement1D elem)
         [Output("propName", "The name of the element's construction property")]
         public static string PrimaryPropertyName(this FrameEdge frameEdge)
         {
-            if (frameEdge.FrameEdgeProperty == null)
+            if (frameEdge == null || frameEdge.FrameEdgeProperty == null)
                 return "";
```
Comment (michaelhoehn) [inline Facade_Engine/Query/PrimaryPropertyName.cs:75]: Can we provide a user warning here at a minimum to inform when there are nulls? Could get confusing to see an empty string if you think you provided all inputs correctly.

File: Facade_Engine/Query/PrimaryPropertyName.cs [hunk 2]
```diff
@@ -82,6 +82,9 @@ public static string PrimaryPropertyName(this FrameEdge frameEdge)
         [Output("propName", "The name of the element's construction property")]
         public static string PrimaryPropertyName(this IEdge edge)
         {
+            if (edge == null)
```
Comment (michaelhoehn) [inline Facade_Engine/Query/PrimaryPropertyName.cs:85]: Same as above.

File: Facade_Engine/Query/PrimaryPropertyName.cs [hunk 3]
```diff
@@ -90,7 +93,7 @@ public static string PrimaryPropertyName(this IEdge edge)
         [Output("propName", "The name of the element's construction property")]
         public static string PrimaryPropertyName(this Panel panel)
         {
-            if (panel.Construction == null)
+            if (panel == null || panel.Construction == null)
```
Comment (michaelhoehn) [inline Facade_Engine/Query/PrimaryPropertyName.cs:96]: Same thing here.

File: Facade_Engine/Query/PrimaryPropertyName.cs [hunk 4]
```diff
@@ -101,7 +104,7 @@ public static string PrimaryPropertyName(this Panel panel)
         [Output("propName", "The name of the element's construction property")]
         public static string PrimaryPropertyName(this Opening opening)
         {
-            if (opening.OpeningConstruction == null)
+            if (opening == null || opening.OpeningConstruction == null)
```
Comment (michaelhoehn) [inline Facade_Engine/Query/PrimaryPropertyName.cs:107]: Same thing here.

**Review by michaelhoehn — CHANGES_REQUESTED** [review]
Some user messages would be nice for the following instances.

File: Facade_Engine/Query/PrimaryPropertyName.cs (hunk 2 above)
Comment (FraserGreenroyd) [reply Facade_Engine/Query/PrimaryPropertyName.cs:85]: This was set up as was, so figured the discipline were happy with an empty string with no warning. If warnings are desired, I am leaving that to the discipline to add either to this PR before merge, or after in their own PR :smile:

File: Facade_Engine/Query/PrimaryPropertyName.cs (hunk 3 above)
Comment (FraserGreenroyd) [reply Facade_Engine/Query/PrimaryPropertyName.cs:96]: This was set up as was, so figured the discipline were happy with an empty string with no warning. If warnings are desired, I am leaving that to the discipline to add either to this PR before merge, or after in their own PR :smile:

File: Facade_Engine/Query/PrimaryPropertyName.cs (hunk 4 above)
Comment (FraserGreenroyd) [reply Facade_Engine/Query/PrimaryPropertyName.cs:107]: This was set up as was, so figured the discipline were happy with an empty string with no warning. If warnings are desired, I am leaving that to the discipline to add either to this PR before merge, or after in their own PR :smile:

**Pull request merged**